pkg/retention/retention.go

```diff
@@ -416,3 +419,107 @@ func DeleteMetricsSegmentData(mmetaFile string, metricSegmentsToDelete map[strin
 		return
 	}
 }
+
+func doInodeBasedDeletion(ingestNodeDir string, maxInodeUsagePercent uint64, deletionWarningCounter int) {
```

Comment:
It would be good to test this function (or part of it). I think it's going to be hard to test like it is though, because a test would have to use a lot of inodes; to make it easier to test, you might want to change this `maxInodeUsagePercent` to a total number of inodes (and perhaps do the calculation of that number in the caller of this function); then to test this you can ingest some data and run this function with a low number of allowed inodes, and make sure the correct segments get deleted